### ISSUE TITLE
MS Excel 2016 / 365: Errors while typing operators in a formula followed by cell coordinates and after pressing enter

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9318]
### Steps to reproduce:
1. Open Excel 2016 / 365
2. In Excel's options enable "allow direct editing in cells" (works only in NVDA alpha-16813,a0b991af and later)
3. In a cell, type =12+a5+a3
4. Press backspace and type another cell
5. Press enter

### Actual behavior:
NVDA raises several com errors while typing operators and new cell coordinates and while deleting them with backspace. It also fails to locate dropdown in previous objects after pressing enter on the cell.
```
IO - inputCore.InputManager.executeGesture (15:16:50.936):
Input: kb(laptop):e
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:16:50.963):
oleacc.AccessibleObjectFromEvent with window 329542, objectID 143 and childID 0: [Error -2147467259] Unbekannter Fehler
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:16:50.973):
oleacc.AccessibleObjectFromEvent with window 329542, objectID 144 and childID 0: [Error -2147467259] Unbekannter Fehler
IO - inputCore.InputManager.executeGesture (15:16:51.359):
Input: kb(laptop):shift+8
IO - speech.speakTypedCharacters (15:16:51.380):
typed word: summe
IO - inputCore.InputManager.executeGesture (15:16:51.661):
Input: kb(laptop):a
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:16:51.736):
oleacc.AccessibleObjectFromEvent with window 329542, objectID 147 and childID 0: [Error -2147467259] Unbekannter Fehler
IO - inputCore.InputManager.executeGesture (15:16:52.053):
Input: kb(laptop):1
IO - inputCore.InputManager.executeGesture (15:16:52.315):
Input: kb(laptop):shift+.
IO - inputCore.InputManager.executeGesture (15:16:53.107):
Input: kb(laptop):a
IO - inputCore.InputManager.executeGesture (15:16:53.648):
Input: kb(laptop):5
DEBUGWARNING - NVDAObjects.UIA.UIA._prefetchUIACacheForPropertyIDs (15:16:54.154):
IUIAutomationElement.buildUpdatedCache failed given IDs of set([30019, 30022, 30086, 30025, 30036, 30070, 30103, 30008, 30009, 30010, 30046, 30079])
ERROR - eventHandler.executeEvent (15:16:54.155):
error executing event: typedCharacter on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionUIA object at 0x0522F190> with extra args of {'ch': u':'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 913, in event_typedCharacter
  File "speech.pyc", line 683, in speakTypedCharacters
  File "api.pyc", line 248, in isTypingProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 787, in _get_isProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1117, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 741, in _getUIACacheablePropertyValue
COMError: (-2146233083, None, (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.UIA.UIA._prefetchUIACacheForPropertyIDs (15:16:54.184):
IUIAutomationElement.buildUpdatedCache failed given IDs of set([30019, 30022, 30086, 30025, 30036, 30070, 30103, 30008, 30009, 30010, 30046, 30079])
ERROR - eventHandler.executeEvent (15:16:54.184):
error executing event: typedCharacter on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionUIA object at 0x0522F190> with extra args of {'ch': u'a'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 913, in event_typedCharacter
  File "speech.pyc", line 683, in speakTypedCharacters
  File "api.pyc", line 248, in isTypingProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 787, in _get_isProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1117, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 741, in _getUIACacheablePropertyValue
COMError: (-2146233083, None, (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.UIA.UIA._prefetchUIACacheForPropertyIDs (15:16:54.210):
IUIAutomationElement.buildUpdatedCache failed given IDs of set([30019, 30022, 30086, 30025, 30036, 30070, 30103, 30008, 30009, 30010, 30046, 30079])
ERROR - eventHandler.executeEvent (15:16:54.210):
error executing event: typedCharacter on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionUIA object at 0x0522F190> with extra args of {'ch': u'5'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 913, in event_typedCharacter
  File "speech.pyc", line 683, in speakTypedCharacters
  File "api.pyc", line 248, in isTypingProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 787, in _get_isProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1117, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 741, in _getUIACacheablePropertyValue
COMError: (-2146233083, None, (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (15:16:54.257):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:16:54.259):
oleacc.AccessibleObjectFromEvent with window 329542, objectID 177 and childID 0: [Error -2147467259] Unbekannter Fehler
IO - inputCore.InputManager.executeGesture (15:16:54.400):
Input: kb(laptop):shift+9
DEBUGWARNING - NVDAObjects.UIA.UIA._prefetchUIACacheForPropertyIDs (15:16:54.415):
IUIAutomationElement.buildUpdatedCache failed given IDs of set([30019, 30022, 30086, 30025, 30036, 30070, 30103, 30008, 30009, 30010, 30046, 30079])
ERROR - eventHandler.executeEvent (15:16:54.417):
error executing event: typedCharacter on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionUIA object at 0x0522F190> with extra args of {'ch': u')'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 913, in event_typedCharacter
  File "speech.pyc", line 683, in speakTypedCharacters
  File "api.pyc", line 248, in isTypingProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 787, in _get_isProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1117, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 741, in _getUIACacheablePropertyValue
COMError: (-2146233083, None, (None, None, None, 0, None))
IO - inputCore.InputManager.executeGesture (15:16:55.181):
Input: kb(laptop):enter
DEBUGWARNING - NVDAObjects.UIA.UIA._prefetchUIACacheForPropertyIDs (15:16:55.220):
IUIAutomationElement.buildUpdatedCache failed given IDs of set([30019, 30022, 30086, 30025, 30036, 30070, 30103, 30008, 30009, 30010, 30046, 30079])
ERROR - eventHandler.executeEvent (15:16:55.221):
error executing event: typedCharacter on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionUIA object at 0x0522F190> with extra args of {'ch': u'\r'}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 913, in event_typedCharacter
  File "speech.pyc", line 683, in speakTypedCharacters
  File "api.pyc", line 248, in isTypingProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 787, in _get_isProtected
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\UIA\__init__.pyc", line 1117, in _get_states
  File "NVDAObjects\UIA\__init__.pyc", line 741, in _getUIACacheablePropertyValue
COMError: (-2146233083, None, (None, None, None, 0, None))
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:16:55.342):
oleacc.AccessibleObjectFromEvent with window 394524, objectID 203 and childID 0: [Error -2147467259] Unbekannter Fehler
DEBUGWARNING - NVDAObjects.window.excel.ExcelBase._getDropdown (15:16:55.551):
Could not locate dropdown list in previous objects
DEBUG - treeInterceptorHandler.update (15:16:55.565):
Adding new treeInterceptor to runningTable: <NVDAObjects.window.excel.ExcelBrowseModeTreeInterceptor object at 0x052537D0>
```

### Expected behavior:
No errors.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA alpha-16813,a0b991af

#### Windows version:
Windows 10 1809 update

#### Name and version of other software in use when reproducing the issue:
MS Excel 2016 / 365

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA?
no, older Alpha do not work with direct cell editing enabled.

**Comment by DrSooom** [comment]
@Adriani90: "Expected behaviour" is missing, even if it's clear for everybody. 😉

**Comment by Adriani90** [comment]
Ok this seems not to occur anymore in 2019.1 Beta 1. Closing as works for me.
